HelenasBank/Controllers/TransferController.cs

```diff
@@ -0,0 +1,78 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using HelenasBank.Models;
+using HelenasBank.Repo;
+using HelenasBank.ViewModels;
+using Microsoft.AspNetCore.Mvc;
+
+namespace HelenasBank.Controllers
+{
+    public class TransferController : Controller
+    {
+        private IBankRepository _repository;
+        public TransferController(IBankRepository repository)
+        {
+            _repository = repository;
+
+        }
+
+        public IActionResult Transfer()
+        {
+            var model = new TransferViewModel()
+            {
+                TransferingAccount = new Models.Account(),
+                ReceivingAccount = new Models.Account()
+            };
+            return View(model);
+        }
+        [HttpPost]
+        public IActionResult Transfer(TransferViewModel model)
+        {
+ 
+
+            if (ModelState.IsValid)
+            {
+                var receivingAccount = new Account();
+
+                var transferingAccount = new Account();
+
+                model.ReceivingAccount.Balance = receivingAccount.Balance;
+                model.TransferingAccount.Balance = transferingAccount.Balance;
+
+                receivingAccount = _repository.GetCustomers().SelectMany(x => x.Accounts.Where(y => y.AccountNo == model.ReceivingAccount.AccountNo)).SingleOrDefault();
+
+                transferingAccount = _repository.GetCustomers().SelectMany(x => x.Accounts.Where(y => y.AccountNo == model.TransferingAccount.AccountNo)).SingleOrDefault();
+
+                if (transferingAccount == null || receivingAccount == null)
+                {
+                    ModelState.AddModelError("", "Account number doesn't exist.");
+                    return View(model);
+                }
+
+
+
+                var result = model.TransferingAccount.TransferFromAccount(model.ReceivingAccount, model.Amount);
```

Comment:
Blir lite knas här. När jag testar så blir balance too low oavsett. Modellen verkar bara innehålla kontonummer, medan t.ex. receivingAccount innehåller allt som behövs. Vill du kolla på det? :)

Reply:
Tack för feedback, testa hur det fungerar nu!

Reply:
Nu kraschar det om man skriver in fel kontonummer, så var säkert därför du hade det i den ordningen från början! :P Men all annan felhantering fungerar, samt unit test osv :)

Reply:
Nu ska det vara fixat!

Reply:
Grymt! Det funkar! Bra jobbat!